### ISSUE TITLE
typescript package tsconfig flags and console output

### ISSUE THREAD

**Issue by MastroLindus** [meteor/meteor#10939]
Some of the issues below are already mentioned in other tickets, but I have the feeling that a proper generic issue is due to give it visibility, as for now the current package is unusable for many users.

1) The typescript package doesn't respect the tsconfig flags, and many of them are hardcoded inside the meteor/babel package.
I understand that meteor needs to control some compilation options in order to ensure that the whole build chain works as expected. Stuff like outDirs, module format, etc.
However:

-  There are many flags that are optional and meteor shouldn't have any saying about them. Stuff like "strict", "noUnusedLocals", etc.

-  Many of these flags directly impacts what kind of libraries we can use. 
We are not allowed to specify "experimentalDecorators" and that can break many libraries that rely on them.
And the fact that "esModuleInterop" is forced to be false in order to optimize the flow with Babel, breaks many libraries that use commonjs or other module systems and that rely on the "import * as " syntax ( moment.js, agenda.js,  sharp and many others)

- In general it is very confusing for the end user that he can specify stuff in his tsconfig.json (and he still might need to do it for IDE integration) but the building process behaves differently

2) No output of the compilation is shown in the console. 
Using in-editor windows is not a solution to this. Most editors only show errors in opened files, and anyway the result could be different as there's no guarantee that the editor TS version is the same as the one used by meteor. 
And in other contexts (like when building and deploying in another environment) it doesn't help anyway. 

What I would expect:

I would expect that meteor would enforce only a very limited amount of flags, only the ones strictly necessary to its own functionality, and they should be documented somewhere. 
For the rest it should try at least a best effort approach in order to respect user-configured flags in tsconfig.json. 

And I would expect to at least have an option to enable typescript's warnings/errors to be visible in the output.

**Comment by afrokick** [comment]
There is a PR for decorators https://github.com/meteor/babel/pull/28

**Comment by MastroLindus** [comment]
not stale

**Comment by MastroLindus** [comment]
not stale

**Comment by perbergland** [comment]
Is there any interest from the maintainers to move the TS plugin in this direction?

There is some recent discussion around using babelrc for setting tsconfig options. but to make the TS compilation part of the build process so we can get proper support for build servers and console diagnostics has not been mentioned and it's unclear to me if there are underlying technical reasons preventing this or if there is something else

**Comment by perbergland** [comment]
There is now a new third-party compiler that has what you are asking for, @MastroLindus.

Please check it out and maybe contribute with some improvements?

Here’s the announcement:
https://forums.meteor.com/t/ann-working-early-version-0-0-3-of-refapp-meteor-typescript-a-brand-new-fast-typescript-compiler-with-tsconfig-json-support-and-diagnostics-output/52943/2

**Comment by perbergland** [comment]
Related link - https://github.com/Meteor-Community-Packages/organization/issues/48

**Comment by zenhack** [comment]
(still relevant)

**Comment by filipenevola** [comment]
Hi @zenhack we have some TypeScript changes on Meteor 1.12 https://github.com/meteor/meteor/pull/11230 (read the changes on this version here https://github.com/meteor/meteor/blob/release-1.12/History.md#changes-1) 

What else do you think is necessary? Let's discuss here.

**Comment by zenhack** [comment]
@filipenevola, the key thing is actually seeing type errors in the console when developing; right now afaik the plugin only does the translation but doesn't actually report the results of type checking. I haven't really dug into this since flipping the switch close to when the plugin was first released, but I don't see anything that seems relevant in the changelog.

**Comment by ruohki** [comment]
the output bothers me as well, my current solution is to have all files open in vscode :| and let intelli sens figure it out

**Comment by afrokick** [comment]
You can run `tsc —noEmit` to check all ts code

**Comment by ruohki** [comment]
Yes, but thats just the same as in vscode

Von meinem iPhone gesendet

**Comment by zenhack** [comment]
Yeah, I've been running tsc manually for the typechecking results, but it's still annoying that the auto code reloading that meteor does in dev mode doesn't do type checking.

**Comment by perbergland** [comment]
My recommendation: use the community typescript package instead (refapp:meteor-typescript). It reads your tsconfig file and shows compilation errors in the console.

**Comment by italojs** [comment]
[Abandoned]  
Please inform us if this occurs in Meteor 3.x.
